api/auth.py

```
@@ -341,13 +53,19 @@ async def get_api_key(request: Request, api_key: str = Depends(api_key_header))
-    key = key_manager.validate_key(api_key)
+    try:
+        # Validate the key
+        key = key_manager.validate_key(api_key)
 
-    # Check rate limit
-    key_manager.check_rate_limit(api_key)
+        # Check rate limit
+        key_manager.check_rate_limit(api_key)
 
-    return key
+        return key
```

Ruff (0.14.10)

Consider moving this statement to an `else` block

(TRY300)